backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/ServerInfoSynchronizer.java

```diff
@@ -41,6 +45,16 @@ public void synchronize(ServerApi serverApi, SonarLintCancelMonitor cancelMonito
     var serverStatus = serverApi.system().getStatus(cancelMonitor);
     var serverVersionAndStatusChecker = new ServerVersionAndStatusChecker(serverApi);
     serverVersionAndStatusChecker.checkVersionAndStatus(cancelMonitor);
-    storage.serverInfo().store(serverStatus);
+    var isMQRMode = serverApi.isSonarCloud() || isMQRMode(serverApi.settings(), serverStatus.getVersion(), cancelMonitor);
```

Comment:
Maybe not a big deal, I'm surprised you save in the storage the result of this condition and not the raw setting from the server. I was expecting the other way around, save the raw setting and have the big condition checked when reading the value from the storage. That said, I think it doesn't really matter

Reply:
I switched to do the other way around and re-introduced the ENUM